### ISSUE TITLE
Fails to build with GCC 14

### ISSUE THREAD

**Issue by sebastic** [shapely/shapely#2081]
As reported in [Debian Bug #1075423](https://bugs.debian.org/1075423) shapely fails to build with GCC 14:
```
src/ufuncs.c: In function ‘polygonize_func’:
src/ufuncs.c:2205:51: error: passing argument 2 of ‘GEOSPolygonize_r’ from incompatible pointer type [-Wincompatible-pointer-types]
 2205 |     GEOSGeometry* ret_ptr = GEOSPolygonize_r(ctx, geoms, n_geoms);
      |                                                   ^~~~~
      |                                                   |
      |                                                   GEOSGeometry ** {aka struct GEOSGeom_t **}
```

See also: https://gcc.gnu.org/gcc-14/porting_to.html#incompatible-pointer-types

**Comment by sebastic** [comment]
gcc-14 is the default compiler in Debian unstable now, the use of `-Wno-error=incompatible-pointer-types` is required until a proper fix for this issue is available.

**Comment by jorisvandenbossche** [comment]
The problem is that the second argument of `GEOSPolygonize_r` is defined as `const GEOSGeometry * const geoms[]`, and not `GEOSGeometry** geoms` (as it is done in some other functions taking an array of geometries, and how we create and pass it in our code).

Not directly sure about the best C way to resolve this.


https://github.com/shapely/shapely/blob/9a8bf73ecf2ef51d684b5866ab5718220b402640/src/ufuncs.c#L2275-L2299

**Comment by musicinmybrain** [comment]
This should already be fixed in `main` by https://github.com/shapely/shapely/pull/1945, which was merged on 2024-12-05 as 05455886750680728dc751dc5888cd02086d908e but never cherry-picked over to `maint-2.0`.

**Comment by jorisvandenbossche** [comment]
Good catch, thanks @musicinmybrain. We should indeed ideally have backported that.

**Comment by jorisvandenbossche** [comment]
> Good catch, thanks @musicinmybrain. We should indeed ideally have backported that.

I did that now for the case we would have to do another 2.0.x bug-fix release, we would at least not forget to include it

**Comment by jorisvandenbossche** [comment]
Given that this is fixed, at least for the upcoming 2.1 release, closing this issue.
